### PR TITLE
Migrating to AWS SDK V3 | Implementation In Analyze Resource and Namespace Monitor

### PR DESCRIPTION
**Pull request by shirady**
### Explain the changes
1. Implementation of AWS SDK V3 In Namespace Monitor.
2. Implementation of AWS SDK V3  In Analyze Resource (use passed region).
3. Omit endpoint parameter for AWS endpoint when using AWS SDK V3.
4. Change the default in the config file to use AWS SDK v3.

### Issues: Fixed #xxx / Gap #xxx
1. none.

### Testing Instructions:
**Analyze Resource**
1. Deploy noobaa on MInikube or Rancher Desktop (see [guide](https://github.com/noobaa/noobaa-operator/blob/master/doc/dev_guide/deply_noobaa_on_minikube_or_rancher_desktop.md)).
_Note: `nb` is an alias that runs the local operator from `build/_output/bin` (alias created by `devenv`)._
2. For backingstore, run:
`nb backingstore create aws-s3 <backing-store-name>` (optional: add the `--region` flag)
`nb backingstore create s3-compatible <backing-store-name>`
`nb diagnostics analyze backingstore <backing-store-name>`
3. For namespacestore, run:
`nb namespacestore create aws-s3 <namespace-store-name>`  (optional: add the `--region` flag)
`nb namespacestore create s3-compatible <namespace-store-name>`
`nb diagnostics analyze namespacestore <namespace-store-name>`
3. For all resources (backingstore and namespacestore), run:
`nb diagnostics analyze resources`
It creates the log files and then deletes the created job.
5. Check the logs of noobaa-core: `kubectl logs -f` look for `test_s3_resource` printings.

Cases:
1 - s3-compatible - Noobaa bucket (with noobaa system as a server, the endpoint was taken from the field `InternalDNS` under `S3 Addresses`).
2 - aws-s3 without flag region on a bucket in a region that is not 'us-east-1' (for example 'us-west-1') and aws-s3 with flag region.

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

**Review by Neon-White — APPROVED** [review]

File: config.js [hunk 1]
```diff
@@ -723,7 +723,7 @@ config.TIERING_TTL_MS = 30 * 60 * 1000; // 30 minutes
 // AWS SDK VERSION     //
 /////////////////////////
 
-config.AWS_SDK_VERSION_3_DISABLED = true;
+config.AWS_SDK_VERSION_3_DISABLED = false;
```
[the remark this answers is not in the document]
Comment (shirady) [reply config.js:726]: It's not too late, I can refactor it (in this PR or in a separate PR it doesn't matter).
I'll share my thoughts so we will find a better suggestion.
I use the suffix DISABLED to indicate that it's a boolean, the suggestion to remove it doesn't indicate what kind of value to expect.
The main use of this variable is in case the new AWS SDK V3 would create a bug and we want to switch back to AWS SDK V2 fast - the main use of it is here:
https://github.com/noobaa/noobaa-core/blob/06d01f2e4c1ac1cd14368f82e550946ec2e5c23b/src/sdk/noobaa_s3_client/noobaa_s3_client.js#L28-L33
Changing it to ENABLED can be a solution for the double negative you're raising, but it would create a negative condition in the function above (`return !config.AWS_SDK_VERSION_3_ENABLED`...).
WDYT?

[the remark this answers is not in the document]
Comment (shirady) [reply config.js:726]: If it is confusing, then it should be changed.
I changed it to `AWS_SDK_VERSION_3_ENABLED`.

File: src/sdk/noobaa_s3_client/noobaa_s3_client.js [hunk 2]
```diff
@@ -89,7 +97,7 @@ function get_requestHandler_with_suitable_agent(endpoint) {
             httpAgent: agent
         });
     } else {
-        throw new Error(`Protocol of ${endpoint} doesn't have a matching agent (valid only for HTTP, HTTPS, WSS only)`);
+        dbg.error(`Protocol of ${endpoint} doesn't have a matching agent (valid only for HTTP, HTTPS, WSS only)`);
```
Comment (dannyzaken) [inline src/sdk/noobaa_s3_client/noobaa_s3_client.js:100]: why did you remove the `throw`, is this a valid state? if yes then we shouldn't log an error. if it's an invalid state it's better to throw an identify it earlier

Comment (shirady) [reply src/sdk/noobaa_s3_client/noobaa_s3_client.js:100]: I removed it because this function meant to get `NodeHttpHandler` and I thought that there might be cases that a S3 client doesn't really need this property and it might be fine, but rethinking it - in the cases where it is needed and someone used it improperly it would create weird errors... I'll throw the error as it was.

File: src/tools/diagnostics/analyze_resource/analyze_resource.js [hunk 3]
```diff
@@ -24,6 +25,7 @@ async function main() {
     try {
         const resource_type = process.env.RESOURCE_TYPE;
         check_supported_resource_type(resource_type);
+        console.info('connection_basic_details', connection_basic_details);
```
Comment (liranmauda) [inline src/tools/diagnostics/analyze_resource/analyze_resource.js:28]: change it to dbg maybe?

**Review by liranmauda — APPROVED** [review]
LGTM

File: src/tools/diagnostics/analyze_resource/analyze_resource.js (hunk 3 above)
Comment (shirady) [reply src/tools/diagnostics/analyze_resource/analyze_resource.js:28]: I prefer leaving it with the console.log/info/error, etc.
As we checked, I didn't use `dbg.log` in this file (it only warps for the process name and time).

**Review by dannyzaken — APPROVED** [review]
